pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/json/ImmutableJsonIndexReader.java

```diff
@@ -300,6 +305,92 @@ private int getDocId(int flattenedDocId) {
     return _docIdMapping.getInt((long) flattenedDocId << 2);
   }
 
+  /**
+   * ImmutableJsonIndexReaderContext holds a cache that is used to accelerate following reads. The cache is specific
```

Comment:
The cache provides speed up when a json path exists multiple times in the group-by clause?

Reply:
Discussed offline, the cache provides speed up when processing subsequent blocks